src/IECoreScene/MeshPrimitiveEvaluator.cpp

```diff
@@ -908,7 +975,18 @@ void MeshPrimitiveEvaluator::closestPointWalk( TriangleBoundTree::NodeIndex node
 
 				if( m_uv.interpolation != PrimitiveVariable::Invalid )
 				{
-					result->m_uv = result->vec2PrimVar( m_uv );
+					Imath::V2f uv[3];
+					triangleUVs( triangleIndex, vertexIds, uv );
+
+					result->m_uv = Imath::V2f(
```

Comment:
I don't fully understand why doing this particular intermediate calculation in float is more of a problem than other intermediate calculations we're doing a float?  There probably is a reason I haven't thought of, but might be good to note that in a comment.  I guess maybe in worst case it could introduce 3 * epsilon worth of error somehow, and your tolerance in triangeContainsPoint is just 2 * epsilon?

If we do need to use double, I wondered about the more succinct version: `result->m_uv = Imath::V2d( uv[0] ) * static_cast< double >( bary[0] ) + Imath::V2d( uv[1] ) * static_cast< double >( bary[1] ) + Imath::V2d( uv[2] ) * static_cast< double >( bary[2] )` ... I think Imath has constructors that support switching types, but maybe it's better to be explicit here for some reason?